modfile.go

```diff
@@ -0,0 +1,69 @@
+package gograveyard
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"strings"
+)
+
+type ModFile struct {
+	Direct   []*Module
+	Indirect []*Module
+}
+
+type Module struct {
+	Path    string
+	Version string
+}
+
+func Parse(data []byte) (*ModFile, error) {
+	scanner := bufio.NewScanner(bytes.NewReader(data))
+
+	var require bool
+	var f ModFile
+
+	for scanner.Scan() {
+		switch scanner.Text() {
+		case "require (":
+			require = true
+
+			continue
+		case ")":
+			require = false
+
+			continue
+		}
+
+		if !require {
+			continue
+		}
+
+		// If in a require section, parse the module
+		// A module is required to have a path and version, the "indirect" comment is optional
+		m := strings.Split(strings.TrimSpace(scanner.Text()), " ")
+		const pathAndVersion = 2
+		// Skip if invalid module, missing required path or version
+		if len(m) < pathAndVersion {
+			continue
```

Comment:
Is it worth putting a "warning" here for now to see what we may end up skipping?
Is there a test case we can add as well?